src/wallet3/db_schema.cpp

```diff
@@ -40,25 +40,35 @@ namespace wallet
     // TODO: table for balance "per account"
     db.exec(
         R"(
-          -- CHECK (id = 0) restricts this table to a single row
           CREATE TABLE metadata (
-            id INTEGER NOT NULL PRIMARY KEY CHECK (id = 0),
-            db_version INTEGER NOT NULL DEFAULT 0,
-            nettype TEXT NOT NULL DEFAULT "testnet",
-            balance INTEGER NOT NULL DEFAULT 0,
-            unlocked_balance INTEGER NOT NULL DEFAULT 0,
-            last_scan_height INTEGER NOT NULL DEFAULT -1,
-            scan_target_hash TEXT NOT NULL,
-            scan_target_height INTEGER NOT NULL DEFAULT 0,
-            output_count INTEGER NOT NULL DEFAULT 0,
-            spend_priv BLOB,
-            spend_pub BLOB,
-            view_priv BLOB,
-            view_pub BLOB
-          );
-
-          -- insert metadata row as default
-          INSERT INTO metadata VALUES (0,0,"testnet",0,0,-1,'',0,0,'','','','');
+            id TEXT PRIMARY KEY NOT NULL,
+            val_numeric INT,
+            val_binary BLOB,
+            val_text TEXT,
+            -- Exactly one val_* must be set:
+            CHECK((val_numeric IS NOT NULL) + (val_binary IS NOT NULL) + (val_text IS NOT NULL) == 1)
+          ) STRICT;
+
+          INSERT INTO metadata(id, val_numeric)
+          VALUES
+            ('db_version', 0),
+            ('balance', 0),
+            ('unlocked_balance', 0),
+            ('last_scan_height', 0),
+            ('scan_target_height', 0),
+            ('output_count', 0);
+
+          INSERT INTO metadata(id, val_text)
+          VALUES
+            ('nettype', 'testnet'),
+            ('scan_target_hash', '');
+
+          INSERT INTO metadata(id, val_binary)
+          VALUES
+            ('spend_priv', x''),
+            ('spend_pub', x''),
+            ('view_priv', x''),
+            ('view_pub', x'');
```

Comment:
Let's leave this off of the initial data, and do the upsert instead of update when we store them (I suspect we'll want other upsertable data in the future, so useful to have it generically and it works for these).